haystack/components/builders/dynamic_prompt_builder.py

```diff
@@ -166,42 +50,35 @@ def run(self, doc_input: str):
     >> 'completion_tokens': 6,
     >> 'total_tokens': 34}}]}}
 
+    Note how in the example above, we can dynamically change the prompt template by providing a new template to the
+    run method of the pipeline. This dynamic prompt generation is in stark contrast to the static prompt generation
+    using `PromptBuilder`, where the prompt template is fixed for the lifetime of the pipeline and cannot be changed
+    for each pipeline run invocation.
+
     """
 
-    def __init__(self, runtime_variables: Optional[List[str]] = None, chat_mode: Optional[bool] = True):
+    def __init__(self, runtime_variables: Optional[List[str]] = None):
         """
         Initializes DynamicPromptBuilder with the provided variable names. These variable names are used to resolve
         variables and their values during pipeline runtime execution. For example, if `runtime_variables` contains
         `documents` your instance of DynamicPromptBuilder will expect an input called `documents`.
         The values associated with variables from the pipeline runtime are then injected into template placeholders
-        of either a ChatMessage or a string template that is provided to the `run` method.
-        See `run` method for more details.
+        of a string template that is provided to the `run` method.
 
         :param runtime_variables: A list of template variable names you can use in chat prompt construction.
         :type runtime_variables: Optional[List[str]]
-        :param chat_mode: A boolean flag to indicate if the chat prompt is being built for a chat-based prompt
-        templating. Defaults to True.
-        :type chat_mode: Optional[bool]
         """
         runtime_variables = runtime_variables or []
 
         # setup inputs
-        if chat_mode:
-            run_input_slots = {"prompt_source": List[ChatMessage], "template_variables": Optional[Dict[str, Any]]}
-        else:
-            run_input_slots = {"prompt_source": str, "template_variables": Optional[Dict[str, Any]]}
-
+        run_input_slots = {"prompt_source": str, "template_variables": Optional[Dict[str, Any]]}
         kwargs_input_slots = {var: Optional[Any] for var in runtime_variables}
         component.set_input_types(self, **run_input_slots, **kwargs_input_slots)
 
         # setup outputs
-        if chat_mode:
-            component.set_output_types(self, prompt=List[ChatMessage])
-        else:
-            component.set_output_types(self, prompt=str)
+        component.set_output_types(self, prompt=str)
 
         self.runtime_variables = runtime_variables
-        self.chat_mode = chat_mode
 
     def to_dict(self) -> Dict[str, Any]:
```

Comment:
This is just doing the same as the default implementation so we can remove this method completely.